### PR TITLE
[2/n] Team Management - remove users from agency

### PR DESCRIPTION
**Pull request by nichelle-hall**
## Description of the change

onClick that sends request to remove user from an agency. 

Recording is a bit long b/c I forgot to click my name the first time!

https://user-images.githubusercontent.com/19961693/214148523-a84a632e-f64c-4d49-a7fb-06fa2b6e9a8f.mov

### Development

**This box MUST be checked by the submitter prior to merging**:
- [x] **Double- and triple-checked that there is no Personally Identifiable Information (PII) being mistakenly added in this pull request**

These boxes should be checked by the submitter prior to merging:
- [ ] Tests have been written to cover the code changed/added as part of this pull request

### Code review

These boxes should be checked by reviewers prior to merging:

- [ ] This pull request has a descriptive title and information useful to a reviewer
- [ ] This pull request has been moved out of a Draft state, has no "Work In Progress" label, and has assigned reviewers
- [ ] Potential security implications or infrastructural changes have been considered, if relevant

### CODE REVIEW

**Comment by lilidworkin** [comment]
@nichelle-hall is this ready for review? Just came across it accidentally but saw it had no reviewers assigned. No worries if not!

**Comment by nichelle-hall** [comment]
> @nichelle-hall is this ready for review? Just came across it accidentally but saw it had no reviewers assigned. No worries if not!

nope! I'm testing end to end right now.

File: publisher/src/components/Settings/AgencySettingsTeamManagement.tsx [hunk 1]
```diff
@@ -120,8 +120,14 @@ export const AgencySettingsTeamManagement: React.FC<{
       setTeam([newMember, ...team]);
       setNameValue("");
       setEmailValue("");
+      agencyStore.inviteUserToAgency({
+        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+        invite_name: name!,
```
Comment (lilidworkin) [inline publisher/src/components/Settings/AgencySettingsTeamManagement.tsx:125]: Why the exclamation point on name? Is that safe?

File: publisher/src/stores/AgencyStore.ts [hunk 2]
```diff
@@ -143,6 +143,53 @@ class AgencyStore {
...
+      method: "DELETE",
+    })) as Response;
+
+    if (response.status !== 200) {
+      showToast({
+        message: `Failed to remove user.`,
+        color: "red",
+        timeout: 4000,
+      });
+      throw new Error("There was an issue with removing the team member.");
```
Comment (lilidworkin) [inline publisher/src/stores/AgencyStore.ts:159]: We should be consistent with whatever language we use in our errors -- do we say "There was an issue" elsewhere? I'm fine with whatever as long as it's consistent

**Review by lilidworkin — COMMENTED** [review]
Looks good @nichelle-hall ! Adding @ivan-kishko as a reviewer since he is the main owner of this file!

**Comment by lilidworkin** [comment]
@nichelle-hall I'm also wondering about the test video -- why didn't the removal work the first time? It sure looks like you clicked your name

File: publisher/src/components/Settings/AgencySettingsTeamManagement.tsx (hunk 1 above)
Comment (ivan-kishko) [reply publisher/src/components/Settings/AgencySettingsTeamManagement.tsx:125]: There will be a check if name and email are not empty, also I am not sure who to ask but we should have some kind of pattern check for email, for example at least 2 symbols before "@", required "@" symbol, some more symbols after "@", and something about domains

**Pull request closed without merging**